### PR TITLE
CVSL-2145 add homeDetentionCurfewActualDate to HdcLicence

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: run-local-with-dev-db.sh [hunk 1]
```diff
@@ -32,6 +32,8 @@ export HMPPS_AUTH_URL=https://sign-in-dev.hmpps.service.justice.gov.uk/auth
 # Make the connection without specifying the sslmode=verify-full requirement
 export SPRING_DATASOURCE_URL='jdbc:postgresql://${DB_SERVER}/${DB_NAME}'
 
+# Add policy v3 enabled flag
+export POLICYV3_ENABLED=false
```
Comment (louisenorris) [inline run-local-with-dev-db.sh:36]: Needed to run the local script after policy v3 toggle added.

File: run-local.sh [hunk 2]
```diff
@@ -60,6 +60,9 @@ export HMPPS_AUTH_URL=https://sign-in-dev.hmpps.service.justice.gov.uk/auth
 # Make the connection without specifying the sslmode=verify-full requirement
 export SPRING_DATASOURCE_URL='jdbc:postgresql://${DB_SERVER}/${DB_NAME}'
 
+# Add policy v3 enabled flag
+export POLICYV3_ENABLED=false
```
Comment (louisenorris) [inline run-local.sh:64]: Needed to run the local script after policy v3 toggle added.

File: run-local-with-dev-db.sh (hunk 1 above)
Comment (andrewrlee) [reply run-local-with-dev-db.sh:36]: we can delete this script but doesn't need to be done now

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceFactory.kt [hunk 3]
```diff
@@ -231,6 +231,7 @@ object LicenceFactory {
     sentenceEndDate = nomisRecord.sentenceExpiryDate,
     licenceStartDate = nomisRecord.getLicenceStartDate(),
     licenceExpiryDate = nomisRecord.licenceExpiryDate,
+    homeDetentionCurfewActualDate = nomisRecord.homeDetentionCurfewActualDate,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceFactory.kt:234]: This should probably be covered by a test

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/HdcLicence.kt [hunk 4]
```diff
@@ -103,6 +103,13 @@ data class HdcLicence(
   @JsonFormat(pattern = "dd/MM/yyyy")
   override val licenceExpiryDate: LocalDate? = null,
 
+  @Schema(
+    description = "The offender’s actual home detention curfew date",
+    example = "06/05/2023",
+  )
+  @JsonFormat(pattern = "dd/MM/yyyy")
+  val homeDetentionCurfewActualDate: LocalDate? = null,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/HdcLicence.kt:111]: **question:**  does this need to be set somewhere?

Comment (louisenorris) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/HdcLicence.kt:111]: Is `homeDetentionCurfewActualDate` not being set in the `createHdc` function in the licence factory?

Comment (louisenorris) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/model/HdcLicence.kt:111]: Added `homeDetentionCurfewActualDate` to `ToModelTransformers.kt`

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**